libfwupd/fwupd-client.c

```diff
@@ -2039,6 +2112,129 @@ fwupd_client_get_remote_by_id (FwupdClient *client,
 	return g_object_ref (remote);
 }
 
+/**
+ * fwupd_client_set_user_agent:
+ * @client: A #FwupdClient
+ * @user_agent: the user agent ID, e.g. `gnome-software/3.34.1`
+ *
+ * Sets the user agent that is used for downloading.
+ *
+ * Since: 1.5.0
+ **/
+void
+fwupd_client_set_user_agent (FwupdClient *client, const gchar *user_agent)
+{
+	FwupdClientPrivate *priv = GET_PRIVATE (client);
+	g_return_if_fail (FWUPD_IS_CLIENT (client));
+	g_return_if_fail (user_agent != NULL);
+	g_free (priv->user_agent);
+	priv->user_agent = g_strdup (user_agent);
```

Comment:
Is it worth some extra checks here that the agent looks like the expected format?

Reply:
is there an expected format? I mean, any non-zero string is okay, e.g. `dave` or `a b c d=e f="g"`

Reply:
But since this will go up to lvfs you can actually mandate a particular format to make it easier to distinguish with client is in use slot.

Reply:
To your point in the comment if you want clients to always have the "name/version" syntax you can mandate that.  But if you don't think this is a useful effort, understand.

Reply:
Aha, of course. I've added a check to see if at least the fwupd version is set, which is the one we really care about from an LVFS point of view. In the process, I solved a mystery, which is fixed in the final commit in the branch.